module/signature/random_beacon_signer_store.go

```diff
@@ -0,0 +1,109 @@
+package signature
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/storage"
+)
+
+type EpochAwareRandomBeaconSignerStore struct {
+	epochLookup module.EpochLookup          // used to fetch epoch counter by view
+	keys        storage.DKGKeys             // used to fetch DKG private key by epoch
+	signers     map[uint64]module.MsgSigner // cache of signers by epoch
+}
+
+func NewEpochAwareRandomBeaconSignerStore(epochLookup module.EpochLookup, keys storage.DKGKeys) *EpochAwareRandomBeaconSignerStore {
+	return &EpochAwareRandomBeaconSignerStore{
+		epochLookup: epochLookup,
+		keys:        keys,
+		signers:     make(map[uint64]module.MsgSigner),
+	}
+}
+
+// GetSigner returns the threshold-signer for signing objects at a
+// given view. The view determines the epoch, which determines the DKG private
+// key underlying the signer.
+// It returns:
+//  - (signer, nil) if DKG was completed in the epoch of the view, signer is not nil
+//  - (nil, protocol.ErrEpochNotCommitted) if no epoch found for given view
```

Comment:
Does the caller care about this case?

Reply:
The caller would likely treat this case the same as a general unexpected exception.